cmake/concepts_ts.cmake

```diff
@@ -0,0 +1,54 @@
+#
+#  Copyright Christopher Di Bella
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+include(FindPackageHandleStandardArgs)
+include(CheckCXXCompilerFlag)
+include(CMakePushCheckState)
+
+message(STATUS "Checking for concept support")
+
+cmake_push_check_state(RESET)
+   if(${CMAKE_CXX_COMPILER_ID} MATCHES "GNU")
```

Comment:
The compiler checks shouldn't be needed because you're already testing support by compiling a test program.

Reply:
Yes, but they take different flags:

* GCC: `-fconcepts`
* Clang: `-std=c++2a -Xclang -fconcepts-ts`
* MSVC: `/experimental:concepts`